### ISSUE TITLE
Incorrect Results

### ISSUE THREAD

**Issue by AASHISHAG** [bedapudi6788/deeppunct#23]
Hi, I tried training `txt2txt `on a German dataset for punctuation correction. Then used `deepcorrect `to punctuate the text, but the results are incorrect. Could you please guide on this?

I passed the training data itself. But output is consistently 3 words. Please guide.

>>> corrector.correct("in keiner dieser debatten wurden diese grundsätze irgendwie bestritten oder angezweifelt")
[[{'sequence': 'in keiner', 'prob': 0.26473349747315716}]]
>>> corrector.correct("es geht mir in diesem kontext darum spielraum zu schaffen für zwei dinge")
[[{'sequence': 'es geht m', 'prob': 0.1473360733543895}]]

**Comment by AASHISHAG** [comment]
@bedapudi6788 I earlier pointed it at `txt2txt` issues. But I think that's not a correct place, therefore opening it here. Could you please guide.

**Comment by bedapudi6788** [comment]
Please add extra info like, how many training samples, validation accuracy, train accuracy, are predictions on train data correct?

**Comment by AASHISHAG** [comment]
Thank you for the response.

Training Samples: ~20 sentences
Validation accuracy: 1.0000
Train accuracy: 1.0000

I just realized that the length of the output is strangely always 9. I also tried changing `max_lenghts=(10, 10))` to `max_lenghts=(50, 50))`, but still the same result.

```
Epoch 00398: val_accuracy did not improve from 1.00000
Epoch 399/400
99/99 [==============================] - 2s 19ms/step - loss: 1.1921e-07 - accuracy: 1.0000 - val_loss: 1.1921e-07 - val_accuracy: 1.0000
Epoch 00399: val_accuracy did not improve from 1.00000
Epoch 400/400
99/99 [==============================] - 2s 19ms/step - loss: 1.1921e-07 - accuracy: 1.0000 - val_loss: 1.1921e-07 - val_accuracy: 1.0000
```

```
>>> corrector.correct('in keiner dieser debatten wurden diese grundsätze irgendwie bestritten oder angezweifelt')
[[{'prob': 0.9999994039536944, 'sequence': 'in keiner'}]]
```

**Comment by AASHISHAG** [comment]
Clearing params and increasing max_lenghts=(200, 200)) worked on training data. Will now train on a big data test and test. Thank you.

@bedapudi6788 : I had read your comment on other issues, where you mentioned that current txt2txt implementation differs a lot from DEMO. Any plans for releasing the Demo code? or Could you guide, where we need to do changes?

**Comment by bedapudi6788** [comment]
we are working on a faster and more accurate punctuation correction at https://github.com/notAI-tech/fastpunct .

The core idea is similar (seq2seq), but @harikodali figured out a way to make the decoding faster. This repo is expected to be released by the end of this month.
